mm2src/coins/utxo.rs

```diff
@@ -1618,16 +1623,23 @@ fn kmd_interest(
     }
 
     // interest stop accruing after 1 year before block 1000000
-    if minutes > 365 * 24 * 60 {
-        minutes = 365 * 24 * 60
+    if minutes > MINUTES_PER_YEAR {
+        minutes = MINUTES_PER_YEAR
     };
     // interest stop accruing after 1 month past 1000000 block
     if height >= 1_000_000 && minutes > 31 * 24 * 60 {
         minutes = 31 * 24 * 60;
     }
-    // next 2 lines ported as is from Komodo codebase
+    // Some of these lines are ported as is from Komodo codebase
     minutes -= 59;
-    let accrued = (value / 10_512_000) * minutes;
+    let mut accrued = (value as f64 * AUR_PER_MINUTE) as u64 * minutes;
```

Comment:
Just curious to know, what's the purpose of introducing floating point arithmetic here, making a conversion from integer to float and vice versa, using `drop_mutability!` macros instead of simple:
```rust
    let accrued = if height >= N_S7_HARDFORK_HEIGHT {
        (value / 10_512_000) * minutes / 500
    } else {
        (value / 10_512_000) * minutes
    };
```